autoload/vebugger.vim

```diff
@@ -268,26 +270,39 @@ function! vebugger#createDebugger(command)
     return l:debugger
 endfunction
 
+if has('timers')
+    function! ReadingTimerCallback(debugger, timer_id)
+        call a:debugger.invokeReading()
+    endfunction
+endif
 
 "Create a debugger and set it as the currently active debugger
 function! vebugger#startDebugger(command)
     call vebugger#killDebugger()
 
     let s:debugger=vebugger#createDebugger(a:command)
 
-    augroup vebugger_shell
-        autocmd!
-        autocmd CursorHold * call s:debugger.invokeReading()
-    augroup END
+    if has('timers')
+        let s:timer_id = timer_start(500, function('ReadingTimerCallback', [s:debugger]), {'repeat': -1})
+    else
+        augroup vebugger_shell
+            autocmd!
+            autocmd CursorHold * call s:debugger.invokeReading()
+        augroup END
+    endif
 
     return s:debugger
 endfunction
 
 "Terminate the currently active debugger
 function! vebugger#killDebugger()
-    augroup vebugger_shell
-        autocmd!
-    augroup END
+    if has('timers') && exists('s:timer_id')
```

Comment:
Use two `if` statements. If this is called before the first debugger was created, it'll run the autogroup cleanup code. Not that it'll damage anything - but if you are going to do things you should do them right.